pkg/pgmodel/querier/querier.go

```diff
@@ -206,10 +209,15 @@ func (q *pgxQuerier) querySingleMetric(metric string, filter metricTimeRangeFilt
 	rows, err := q.conn.Query(context.Background(), sqlQuery, values...)
 	if err != nil {
 		if e, ok := err.(*pgconn.PgError); ok {
-			// If we are getting undefined table error, it means the metric we are trying to query
-			// existed at some point but the underlying relation was removed from outside of the system.
-			if e.Code == pgerrcode.UndefinedTable {
+			switch e.Code {
+			case pgerrcode.UndefinedTable:
+				// If we are getting undefined table error, it means the metric we are trying to query
+				// existed at some point but the underlying relation was removed from outside of the system.
 				return nil, nil, fmt.Errorf(errors.ErrTmplMissingUnderlyingRelation, mInfo.TableSchema, mInfo.TableName)
+			case pgerrcode.UndefinedColumn:
+				// If we are getting undefined column error, it means the column we are trying to query
+				// does not exist in the metric table so we return empty results.
```

Comment:
what's the rationale behind empty result and not an error? Wouldn't the error be more helpful to the user?

Reply:
Treating missing column as a missing series and returning empty result. Will add a comment.